core/src/filesystem/vfs.cc

```diff
@@ -420,8 +474,12 @@ Status VFS::read(
     return LOG_STATUS(
         Status::VFSError("Cannot read from file; File does not exist"));
 
-  if (uri.is_posix()) {
+  if (uri.is_file()) {
+#ifdef _WIN32
+    return win32::read_from_file(uri.to_path(), offset, buffer, nbytes);
```

Comment:
I changed `posix::read_from_file` and `posix::write_to_file` to `posix::read` and `posix::write` respectively. Perhaps we can do the same for `win32` for consistency.